src/mono/netcore/sample/wasm/browser/Makefile

```diff
@@ -7,12 +7,16 @@ else
 DOTNET_Q_ARGS=--nologo
 endif
 
+ifneq ($(AOT),)
+DOTNET_RUN_AOT_COMPILATION_ARGS=/p:RunAOTCompilation=true
+endif
+
 CONFIG?=Release
 
 all: build
 
 build:
-	$(DOTNET) build $(DOTNET_Q_ARGS) /p:TargetArchitecture=wasm /p:TargetOS=Browser /p:Configuration=$(CONFIG) WasmSample.csproj
+	EMSDK_PATH=$(realpath $(TOP)/src/mono/wasm/emsdk) $(DOTNET) publish $(DOTNET_Q_ARGS) $(DOTNET_RUN_AOT_COMPILATION_ARGS) /p:Configuration=$(CONFIG) /p:TargetArchitecture=wasm /p:TargetOS=Browser $(MSBUILD_ARGS) WasmSample.csproj
```

Comment:
Does this (realpath) work on OS/X without setting SHELL to a bash version newer than what it comes with?

Reply:
we should only set EMSDK_PATH if it's not yet set (so I can use my own emsdk). If you do that as a Makefile variable you can easily resolve it to a full path with the `abspath` Makefile function:

```
EMSDK_PATH?=$(abspath $(TOP)/src/mono/wasm/emsdk)
```